browser/brave_content_browser_client.cc

```diff
@@ -1267,6 +1272,21 @@ BraveContentBrowserClient::CreateThrottlesForNavigation(
       throttles.push_back(std::move(ai_chat_throttle));
     }
   }
+
+  if (Profile::FromBrowserContext(context)->IsRegularProfile()) {
```

Comment:
This moves the logic for determining which profiles to use out of ai_chat. Normally we would do a null check on `AIChatServiceFactory::GetForBrowserContext`, is this meant to be more restrictive than that? If so, it should still be a helper that keeps the logic inside ai_chat

Reply:
This check follows existing code, I think it's before AIChatServiceFactory exists, I will use that instead.

Reply:
removed this in 6cacbb06b4c7421bce8287978c0cd206d069dc0c